hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java

```diff
@@ -303,15 +304,17 @@ public void start() {
           datanodeDetails.setPort(DatanodeDetails.newPort(HTTPS,
                   httpServer.getHttpsAddress().getPort()));
         }
+        serviceRuntimeInfo.setHttpPort(String.valueOf(httpServer.getHttpAddress().getPort()));
       } catch (Exception ex) {
         LOG.error("HttpServer failed to start.", ex);
       }
 
-
       clientProtocolServer = new HddsDatanodeClientProtocolServer(
           datanodeDetails, conf, HddsVersionInfo.HDDS_VERSION_INFO,
           reconfigurationHandler);
 
+      serviceRuntimeInfo.setRpcPort(String.valueOf(clientProtocolServer.getClientRpcAddress().getPort()));
```

Comment:
The datanode has multiple RPC ports for various purposes, see [here](https://github.com/apache/ozone/blob/624bede619004a6a2be5fe185abd40c2999ae3c1/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java#L896). This one is specifically the one used by the client for config reload. Do other components have an RPC port metric, and do they add more information about what this port is used for?

Reply:
Thank you very much for your question! 

I hope to present the RPC port needed for client connections, which might be more appropriately named `ClientRpcPort`.  I should add some comments to make the code clearer. 

- `OM` displays the `RpcPort (9862)` in JMX.  
https://github.com/apache/ozone/blob/d3899d2cc3912fe47314a6af0c7bceee89114789/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMMXBean.java#L28-L33

- `SCM` currently displays `DatanodeRpcPort (9861)` and `ClientRpcPort (9860)` in JMX.
https://github.com/apache/ozone/blob/d3899d2cc3912fe47314a6af0c7bceee89114789/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMMXBean.java#L34-L50